src/cpp/Agent.cpp

```diff
@@ -36,14 +31,193 @@ static dds::xrce::ClientKey raw_to_clientkey(uint32_t key)
     return client_key;
 }
 
+struct RefRep
+{
+    const char* ref;
+};
+
+struct XmlRep
+{
+    const char* xml;
+};
+
+template<Agent::ObjectKind object_kind, typename U, typename T>
+void fill_object_variant(
+        T parent_id,
```

Comment:
I like it. In fact, if in a future update, an entity requires more than one parent, it could also get with a std::pair, or std::tuple of parents. :D

Reply:
👍